ecllibrary/std/Type.ecl

```diff
@@ -0,0 +1,241 @@
+    end;
+    
+    /*
+     * Dynamic translation to arbitrary record structure
+     */
+    EXPORT outrec translateRow(ROW inrow) := externals.translateRow(outrec, inrow);
+    EXPORT streamed dataset(outrec) translate(streamed dataset indata) := externals.translate(indata);
+    
+    /**
+     * Returns the binary type metadata structure for a record.
+     * 
+     * @return           A binary representation  of the type information
+     */
+    EXPORT DATA getBinaryTypeInfo() := externals.serializeRecordType(outrec);
+    /**
+     * Returns the json type metadata structure for a record.
+     * 
+     * @return           A json representation  of the type information
+     */
+    EXPORT STRING getJsonTypeInfo() := externals.dumpRecordType(outrec);
+     
+    EXPORT SET OF STRING fieldNames() := externals.getFieldNames(outrec);
+    EXPORT UNSIGNED4 numFields() := externals.getNumFields(outrec);
+    
+    /**
+     * Returns a serialized copy of a row.
+     * @param  inrow  Input row, in arbitrary layout. This will be translated to output layout then serialized.
+     * @return        Serialized data for the resulting output row
+     */
+    EXPORT DATA serialize(ROW inrow) := externals.serializeRow(outrec, inrow);
+
+    /**
+     * Deserializes serialized row to an ECL row in specified layout.
+     * @param  rowdata Data to deserialize. This should have been created using serialize() with matching output format, or results will be unpredictable.
+     * @return         Deserialized data for the resulting output row
+     */
+    EXPORT outrec deserialize(DATA rowdata) := externals.deserializeRow(outrec, rowdata);
+    
+    /**
+     * Functions to access individual fields within a serialized row without having to deserialize the whole row
+     * @param  parcel    Parcel to access. This should have been created using serialize() with matching output format, or results will be unpredictable.
+     * @param  fieldname Name of field to access.
+     */
+    EXPORT deserializedField(DATA parcel, STRING fieldname) := MODULE
+      EXPORT STRING   readString() := externals.readSerializedString(outrec, parcel, externals.getFieldNum(outrec, fieldname));;
+      EXPORT INTEGER  readInt() := externals.readSerializedInt(outrec, parcel, externals.getFieldNum(outrec, fieldname));
+      EXPORT REAL8    readReal() := externals.readSerializedReal(outrec, parcel, externals.getFieldNum(outrec, fieldname));
+      EXPORT BOOLEAN  readBool() := externals.readSerializedBool(outrec, parcel, externals.getFieldNum(outrec, fieldname));
+      EXPORT UTF8     readUTF8() := externals.readSerializedUtf8(outrec, parcel, externals.getFieldNum(outrec, fieldname));
+      EXPORT DATA     readData() := externals.readSerializedData(outrec, parcel, externals.getFieldNum(outrec, fieldname));
+    END;
```

Comment:
Consider adding a default value parameter for the value returned when field is not of matching type or cannot be coerced into matching type.